pyro/infer/autoguide/guides.py

```diff
@@ -1314,6 +1328,19 @@ def _setup_prototype(self, *args, **kwargs):
             numel[name] = init_loc.numel()
             init_locs[name] = init_loc.reshape(-1)
 
+        # Instantiate conditionals and dependencies as dictionaries.
+        if not isinstance(self.conditionals, dict):
+            self.conditionals = {name: self.conditionals for name in init_locs}
+        if not isinstance(self.dependencies, dict):
+            model_dependencies = poutine.block(get_dependencies)(
+                self._original_model[0], args, kwargs,
+            )
+            guide_dependencies = model_dependencies  # FIXME
```

Comment:
@eb8680 next task is to replace this with the algorithm in (Webb et al. 2018), which will require additional tracing of observed sites.